### PR TITLE
Bjorncs/perf service dump

### PR DESCRIPTION
**Pull request by bjorncs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/maintenance/servicedump/VespaServiceDumperImplTest.java [hunk 1]
```diff
@@ -37,35 +38,41 @@
 
     private static final String HOSTNAME = "host-1.domain.tld";
 
+    private Path tmpDirectory;
+    private FileSystem fileSystem;
+
+    @BeforeEach
+    void create_test_file_system() throws IOException {
+        // Create temporary directory in container
+        fileSystem = TestFileSystem.create();
```
Comment (hakonhall) [inline node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/maintenance/servicedump/VespaServiceDumperImplTest.java:47]: These two can be initialized above, and they can be made `final`.

**Comment by bjorncs** [comment]
@hakonhall PTAL

**Review by hakonhall — APPROVED** [review]
The idea is to tackle config id -> possibly service name or process name later?

File: container-core/src/main/sh/find-pid [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+if ! [[ "${pid}" =~ ^[0-9]+$ ]]; then
+  echo "Could not find valid pid for '${config_id}' (pid='${pid}')" >&2
+  exit 1
+fi
+if [ -z "$(ps -p ${pid} -o pid=)" ]; then
+  echo "Could not find process for '${pid}'" >&2
+  exit 1
+fi
+echo $pid
+exit 0
```
[the remark this answers is not in the document]
Comment (bjorncs) [reply container-core/src/main/sh/find-pid:100]: Thanks for the tip 👍  Still plenty to learn about shell scripting...

**Comment by bjorncs** [comment]
> The idea is to tackle config id -> possibly service name or process name later?

Yes, that will be a future improvement. It's scope is too big to do now.

**Pull request merged**